src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
       )?.label as string) ?? languageOptions[0].label,
   };
 
-  const handleLanguageChange = (newLanguage: EuiComboBoxOptionOption[]) => {
-    const queryLanguage = newLanguage[0].value as string;
-    props.onSelectLanguage(queryLanguage);
-    uiService.Settings.setUserQueryLanguage(queryLanguage);
+  const handleLanguageChange = (newLanguage: any) => {
+    props.onSelectLanguage(newLanguage);
+    uiService.Settings.setUserQueryLanguage(newLanguage);
   };
 
-  uiService.Settings.setUserQueryLanguage(props.language);
```

Comment:
this was here to ensure the initial state has the correct language the first time. likely this might be causing a bug where if you select another language like PPL. and reload the page the actually query language is not the right value so it fails the search request

Reply:
i selected PPL, and then after refreshing and reloading the page, the query language still remains PPL. Were there other scenarios that make this line necessary?

Reply:
added the line back